sdk/src/main/java/com/mapbox/maps/renderer/RendererError.kt

```diff
@@ -0,0 +1,100 @@
+package com.mapbox.maps.renderer
+
+import com.mapbox.maps.MapView
+import javax.microedition.khronos.egl.EGL10
+import javax.microedition.khronos.egl.EGL11
+
+/**
+ * Typed enum wrapping up all the EGL error codes represented as ints.
+ */
+enum class RendererError {
+  /**
+   * Error code returned when no matching EGL config was found meaning that map rendering is not supported on given device.
+   * This could be also checked with static method [MapView.isRenderingSupported] prior to creating the [MapView].
+   */
+  NO_VALID_EGL_CONFIG_FOUND,
+  /**
+   * Represents [EGL10.EGL_BAD_ALLOC].
+   *
+   * Could be thrown when creating context, surface or display and system is running low on resources.
+   * Best action will be try to release Android resources and re-create the [MapView].
+   */
+  EGL_BAD_ALLOC,
+  /**
+   * Represents [EGL10.EGL_NOT_INITIALIZED].
+   */
+  EGL_NOT_INITIALIZED,
+  /**
+   * Represents [EGL10.EGL_BAD_ACCESS].
+   */
+  EGL_BAD_ACCESS,
+  /**
+   * Represents [EGL10.EGL_BAD_ATTRIBUTE].
+   */
+  EGL_BAD_ATTRIBUTE,
+  /**
+   * Represents [EGL10.EGL_BAD_CONFIG].
+   */
+  EGL_BAD_CONFIG,
+  /**
+   * Represents [EGL10.EGL_BAD_CONTEXT].
+   */
+  EGL_BAD_CONTEXT,
+  /**
+   * Represents [EGL10.EGL_BAD_CURRENT_SURFACE].
+   */
+  EGL_BAD_CURRENT_SURFACE,
+  /**
+   * Represents [EGL10.EGL_BAD_DISPLAY].
+   */
+  EGL_BAD_DISPLAY,
+  /**
+   * Represents [EGL10.EGL_BAD_MATCH].
+   */
+  EGL_BAD_MATCH,
+  /**
+   * Represents [EGL10.EGL_BAD_NATIVE_PIXMAP].
+   */
+  EGL_BAD_NATIVE_PIXMAP,
+  /**
+   * Represents [EGL10.EGL_BAD_NATIVE_WINDOW].
+   */
+  EGL_BAD_NATIVE_WINDOW,
+  /**
+   * Represents [EGL10.EGL_BAD_PARAMETER].
+   */
+  EGL_BAD_PARAMETER,
+  /**
+   * Represents [EGL10.EGL_BAD_SURFACE].
+   */
+  EGL_BAD_SURFACE,
+  /**
+   * Represents [EGL11.EGL_CONTEXT_LOST].
+   */
+  EGL_CONTEXT_LOST,
+
+  /**
+   * Represents unknown error that we could not map to any existing EGL error code.
+   */
+  EGL_UNKNOWN
+}
+
+internal fun Int?.mapEglErrorCode(): RendererError {
```

Comment:
I personally don't like extensions on basic type - could be imported accidentally in the wrong context. 
IMHO static method `fromInt` in the enum itself would equally clear

Reply:
Aaaand actually if you'll add int as a member of the enum constant itself - this `when` will be unneeded, will be able to just iterate through values and pick the right one in a simple loop

Reply:
revisited whole class, PTAL again